File: C++/프로그래머스/0/181894. 2의 영역/2의 영역.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr) {
    int start = -1;
    vector<int> answer = {-1};
    for(int i = 0; i < arr.size(); i++) {
        if(arr[i] == 2) {
            if(start < 0) {
                start = i;
                answer.clear();
            }
            for(int j = start; j <= i; j++) {
                answer.push_back(arr[j]);
            }
            start = i + 1;
        }
    }
    return answer;
}

File: Python3/프로그래머스/0/181894. 2의 영역/2의 영역.py
def solution(arr):
    start, end = len(arr), -1
    for i in range(0, len(arr)):
        if arr[i] == 2:
            start = min(start, i)
            end = max(end, i)
    return arr[start:end + 1] if end >= 0 else [-1]